README.md

```diff
@@ -1,346 +1,374 @@
-- IAM sample - [Source code][pubsub_iam_1] | [Documentation][pubsub_iam_2]
+    Windows (PowerShell):
 
-## Google Cloud Speech
+        $env:GOOGLE_APPLICATION_CREDENTIALS="/path/to/service_account_file.json"
 
-- Recognition sample - [Source code][speech_1] | [Documentation][speech_2]
+    __Note for code running on GCE, GAE, or other environments:__
 
-## Google Cloud Storage
+    On Google App Engine, the credentials should be found automatically.
 
-- Auth sample - [Source code][storage_1] | [Documentation][storage_2]
+    On Google Compute Engine, the credentials should be found automatically, but require that
+    you create the instance with the correct scopes.
 
-## Google Cloud Vision
+        gcloud compute instances create --scopes="https://www.googleapis.com/auth/cloud-platform,https://www.googleapis.com/auth/compute,https://www.googleapis.com/auth/compute.readonly" test-instance
 
-- Face detection - [Source code][vision_1] | [Documentation][vision_2]
-- Label detection - [Source code][vision_3] | [Documentation][vision_4]
-- Landmark detection - [Source code][vision_5] | [Documentation][vision_6]
-- Text detection - [Source code][vision_7] | [Documentation][vision_8]
+    If you did not create the instance with the right scopes, you can still
+    upload a JSON service account and set `GOOGLE_APPLICATION_CREDENTIALS`
+    as described.
 
-## Google Prediction API
+    Read more about [Google Cloud Platform Authentication][gcp_auth].
 
-- Hosted Models sample - [Source code][predictionapi_1] | [Documentation][predictionapi_2]
+[node]: https://nodejs.org/
+[auth_command]: https://cloud.google.com/sdk/gcloud/reference/beta/auth/application-default/login
+[gcp_auth]: https://cloud.google.com/docs/authentication#projects_and_resources
 
-## Other Example Apps
+### How to run a sample
 
-- nodejs-getting-started - [Source code][nodejs_1] | [App Engine Tutorial 1][nodejs_2] | [App Engine Tutorial 2][nodejs_3]
-- gcloud-node-todos - [Source code][todos_1]
-- gitnpm - [Source code][gitnpm_1]
-- gcloud-kvstore - [Source code][kvstore_1]
+1. Change directory to one of the sample folders, e.g. `bigquery`:
 
-## More information
+        cd bigquery/
 
-- [Getting started with nodejs on Google Cloud](http://cloud.google.com/nodejs/)
-- See our other [Google Cloud Platform github repos](https://github.com/GoogleCloudPlatform) for sample applications and scaffolding for other frameworks and use cases.
-- [Using the `gcloud` npm module](https://googlecloudplatform.github.io/gcloud-node/#/)
-- [Logging to Google Cloud with Winston](https://github.com/GoogleCloudPlatform/winston-gae)
+1. Install the sample's dependencies (see the sample's README for details):
 
-## Contributing
+        npm install
 
-Contributions welcome!
+1. Run the sample:
 
-See [CONTRIBUTING.md](https://github.com/GoogleCloudPlatform/nodejs-docs-samples/blob/master/CONTRIBUTING.md)
+        node sample_file.js [args]...
 
-### Running the tests
+### How to run the tests
```

Comment:
It might be worthwhile to consider putting this into a separate CONTRIBUTING.md file.